composer/callbacks/checkpoint_saver.py

```diff
@@ -45,28 +56,47 @@ def checkpoint_periodically(interval: Union[str, int, Time]) -> Callable[[State,
 
     if interval.unit == TimeUnit.EPOCH:
         save_event = Event.EPOCH_CHECKPOINT
-    elif interval.unit == TimeUnit.BATCH:
+    elif interval.unit in {TimeUnit.BATCH, TimeUnit.TOKEN, TimeUnit.SAMPLE}:
         save_event = Event.BATCH_CHECKPOINT
     else:
         raise NotImplementedError(
-            f'Unknown checkpointing interval: {interval.unit}. Must be TimeUnit.EPOCH or TimeUnit.BATCH.')
+            f'Unknown checkpointing interval: {interval.unit}. Must be TimeUnit.EPOCH, TimeUnit.BATCH, TimeUnit.TOKEN, or TimeUnit.SAMPLE.'
+        )
+
+    last_value_checkpointed = 0
+    if starting_timestamp is not None:
+        last_value_checkpointed = starting_timestamp.get(interval.unit)
```

Comment:
Is there a simpler way we can propagate this information besides adding something to state and the logic in checkpointing? What if we just call `checkpoint_periodically` but don't `will_actually_checkpoint=False` on after_load (and use timestamp instead?)

Reply:
hmm i think that works, thanks, let me test it.

Reply:
wait no, the addition to state was for eval, checkpointing doesn't need it. `will_actually_checkpoint` was added because EMA calls this funciton but then doesn't write a checkpoint